src/libraries/System.Memory/tests/SequenceReader/BasicTests.cs

```diff
@@ -646,6 +648,89 @@ public void AdvanceTo_AdvancePast()
             }
         }
 
+        [Fact]
+        public void AdvanceTo_End()
```

Comment:
What happens if the last 1-3 segment (of the sequence) are empty buffers and we call `AdvanceToEnd`? Similarly, what happens to `UnreadSequence` if we haven't read those empty segments?

For example, let's say `|` indicates how far we have read, and the blocks are the segments in the sequence.
[data] [data] ... | [] [] []

Is this scenario worth testing?

Reply:
`AdvanceToEnd` set a bunch of props to defaults and `UnreadSequence` is a simple usage of underlying Slice so I don't see subtle problems, btw I'll add some tests to cover those, thank's!

Reply:
Added some tests take a look and let me know if are enough